core/exception/hierarchy_spec.rb

```diff
@@ -0,0 +1,69 @@
+require_relative '../../spec_helper'
+
+describe "Exception" do
+  it "has the right class hierarchy" do
+    def save_pairs(parent, hash, pairs)
```

Comment:
Could you make this a lambda and move it below the `hierarchy = {`?

Reply:
Tracking the `pairs` explicitly seems extra complexity, I would just use `should` inside this lambda.

Reply:
Done (I think).  Learned about lambdas.  (Learning is my "pay.")